src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -488,6 +509,14 @@ private static void ThrowIfCURLMError(int error)
             }
         }
 
+        private static bool AreEqualNetworkCredentials(NetworkCredential credential1, NetworkCredential credential2)
+        {
+            Debug.Assert(credential1 != null && credential2 != null, "arguments are non-null in network equality check");
+            return credential1.UserName == credential2.UserName &&
+                credential1.Domain == credential2.Domain &&
+                string.Equals(credential1.Password, credential2.Password, StringComparison.Ordinal);
```

Comment:
Equals method in ```NetworkCredential``` is not overloaded. I copied the condition from [NetworkCredential.cs](https://github.com/dotnet/corefx/blob/master/src/System.Net.Primitives/src/System/Net/NetworkCredential.cs#L164)